lib/fog/openstack.rb

```diff
@@ -310,7 +311,8 @@ def self.retrieve_tokens_v2(options, connection_options = {})
         :headers  => {'Content-Type' => 'application/json'},
         :body     => Fog::JSON.encode(request_body),
         :method   => 'POST',
-        :path     => (uri.path and not uri.path.empty?) ? uri.path : 'v2.0'
+        :path     => (uri.path and not uri.path.empty?) ? uri.path : 'v2.0',
+        :omit_default_port => omit_default_port
```

Comment:
I think it might be better to not even list the :omit_default_port if it's nil, so we don't change the current behavior

Could you store the request into {}

then 

request[:omit_default_port] = omit_default_port unless omit_default_port.nil? # that should put here true or false
response = identity_v2_connection.request(request)

same for the keystone v3